flaml/autogen/agent/assistant_agent.py

```diff
@@ -0,0 +1,45 @@
+from .agent import Agent
+from flaml.autogen.code_utils import DEFAULT_MODEL
+from flaml import oai
+
+
+class AssistantAgent(Agent):
+    """(Experimental) Assistant agent, able to suggest code blocks."""
+
+    DEFAULT_SYSTEM_MESSAGE = """You are a helpful AI assistant.
+    In the following cases, suggest python code (in a python coding block) or shell script (in a sh coding block) for the user to execute. You must indicate the script type in the code block.
+    1. When you need to ask the user for some info, use the code to output the info you need, for example, browse or search the web, download/read a file.
+    2. When you need to perform some task with code, use the code to perform the task and output the result. Finish the task smartly. Solve the task step by step if you need to.
+    If you want the user to save the code in a file before executing it, put # filename: <filename> inside the code block as the first line. Don't include multiple code blocks in one response. Do not ask users to copy and paste the result. Instead, use 'print' function for the output when relevant. Check the execution result returned by the user.
+    If the result indicates there is an error, fix the error and output the code again. Suggeset the full code instead of partial code or code changes.
+    Reply "TERMINATE" in the end when everything is done.
+    """
+
+    DEFAULT_CONFIG = {
+        "model": DEFAULT_MODEL,
+    }
+
+    def __init__(self, name, system_message=DEFAULT_SYSTEM_MESSAGE, **config):
```

Comment:
I think it is better to explicitly declare all necessary configs in parameters instead of in `**config`, or to add more documents to inform what configs are needed, for example, `config_list`

Reply:
Updated. Could you check whether it works?